core/src/main/java/org/apache/druid/data/input/impl/JsonInputFormat.java

```diff
@@ -109,9 +144,13 @@ public boolean isSplittable()
   @Override
   public InputEntityReader createReader(InputRowSchema inputRowSchema, InputEntity source, File temporaryDirectory)
   {
-    return this.lineSplittable ?
-           new JsonLineReader(inputRowSchema, source, getFlattenSpec(), objectMapper, keepNullColumns) :
-           new JsonReader(inputRowSchema, source, getFlattenSpec(), objectMapper, keepNullColumns);
+    if (this.lineSplittable || this.assumeNewlineDelimited) {
```

Comment:
When would `lineSplitable` be false, but the the data is actually newLineDelimited? Does this situation arise  when you have a batch of new line delimited json events being ingested at one time, for example?

Reply:
`lineSplittable` is used as an indicator from the task implementation as to whether it expects the input to be newline delimited (and thus use either JsonLineReader or JsonReader). Prior to this patch, batch tasks would always use newline delimited (JsonLineReader) and streaming tasks would always use the multiline parser (JsonReader).

With this patch, batch tasks would still always use newline delimited, while for streaming tasks the behavior is configurable (controlled by the new asssumeNewlineDelimited property)